### PR TITLE
Clientpr/recipe steps panel

### PR DESCRIPTION
**Pull request by dlakshmi123**
# TITLE

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

### CODE REVIEW

File: app/cdap/components/DrawerWidget/DrawerWidgetHeading/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Typography } from '@material-ui/core';
+import React from 'react';
+import { useStyles } from 'components/DrawerWidget/styles';
```
Comment (dlakshmi123) [inline app/cdap/components/DrawerWidget/DrawerWidgetHeading/index.tsx]: remove useStyles as we are not using

File: app/cdap/components/DrawerWidget/iconStore.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import { blue } from '@material-ui/core/colors';
+
+export const BackIcon = (
+  <svg width="13" height="23" viewBox="0 0 13 23" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dlakshmi123) [inline app/cdap/components/DrawerWidget/iconStore.tsx]: move this file to IconStore file

File: app/cdap/components/DrawerWidget/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  headingText,
+  openDrawer,
+  showDivider,
+  headerActionTemplate,
+  children,
+  closeClickHandler,
+  showBackIcon,
+  anchor,
+}: IDrawerWidgetProps) {
+  const classes = useStyles();
```
Comment (dlakshmi123) [inline app/cdap/components/DrawerWidget/index.tsx]: use styled components

File: app/cdap/components/GridTable/index.tsx [hunk 4]
```diff
@@ -39,24 +35,112 @@ import React, { useEffect, useState } from 'react';
...
 export default function GridTable() {
   const { wid } = useParams() as IRecords;
   const params = useParams() as IRecords;
   const classes = useStyles();
+  const { dataprep } = DataPrepStore.getState();
+  const [isFirstWrangle, setIsFirstWrangle] = useState<boolean>(false);
+  const [connectorType, setConnectorType] = useState<string | null>(null);
+  const [openDirectivePanel, setDirectivePanel] = useState<boolean>(true);
+  const [toaster, setToaster] = useState({
+    open: false,
```
Comment (dlakshmi123) [inline app/cdap/components/GridTable/index.tsx:70]: add type

File: app/cdap/components/GridTable/index.tsx [hunk 5]
```diff
@@ -39,24 +35,112 @@ import React, { useEffect, useState } from 'react';
...
   const [headersNamesList, setHeadersNamesList] = useState<IHeaderNamesList[]>([]);
-  const [rowsDataList, setRowsDataList] = useState([]);
+  const [rowsDataList, setRowsDataList] = useState<IRowData[]>([]);
+  const [showRecipePanel, setShowRecipePanel] = useState<boolean>(false);
   const [gridData, setGridData] = useState({} as IExecuteAPIResponse);
-  const [missingDataList, setMissingDataList] = useState([]);
-  const [invalidCountArray, setInvalidCountArray] = useState([
+  const [missingDataList, setMissingDataList] = useState<IMissingListData[]>([]);
+  const [workspaceName, setWorkspaceName] = useState<string>('');
+  const [directiveFunction, setDirectiveFunction] = useState('');
```
Comment (dlakshmi123) [inline app/cdap/components/GridTable/index.tsx]: add type

File: app/cdap/components/GridTable/index.tsx [hunk 6]
```diff
@@ -39,24 +35,112 @@ import React, { useEffect, useState } from 'react';
...
+            headers: response.headers,
+            types: response.types,
+            ...gridParams,
+          },
+        });
+        setLoading(false);
+        setGridData(response);
+        setToaster({
+          open: true,
+          message: `Transformation successfully added`,
```
Comment (dlakshmi123) [inline app/cdap/components/GridTable/index.tsx]: translation text

File: app/cdap/components/GridTable/index.tsx [hunk 7]
```diff
@@ -39,24 +35,112 @@ import React, { useEffect, useState } from 'react';
...
+  const updateDataTranformation = (wid: string) => {
+    const payload: IParams = {
+      context: params.namespace as string,
+      workspaceId: wid,
+    };
+    getWorkSpaceData(payload, wid);
+    setIsFirstWrangle(false);
+  };
+
+  const isParsingPanel =
```
Comment (dlakshmi123) [inline app/cdap/components/GridTable/index.tsx]: if not using remove it

File: app/cdap/components/RecipeSteps/RecipeHeaderActionTemplate/index.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import React from 'react';
+import { useStyles } from 'components/RecipeSteps/styles';
+import { DownloadIcon, KebabIcon } from 'components/RecipeSteps/iconStore';
+import { Box } from '@material-ui/core';
+
+export default function() {
+  const classes = useStyles();
+
+  return (
+    <Box className={classes.downloadMenuActionWrapper} data-testid="header-action-template-parent">
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/RecipeHeaderActionTemplate/index.tsx]: use styled components

File: app/cdap/components/RecipeSteps/RecipeStepsEmptyScreen/index.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import React, { Fragment } from 'react';
+import { useStyles } from 'components/RecipeSteps/styles';
+import T from 'i18n-react';
+import { InfoGraphicData } from 'components/RecipeSteps/iconStore';
+
+export default function() {
+  const classes = useStyles();
+
+  return (
+    <Container className={classes.emptyScreenStyles} data-testid="recipe-steps-empty-screen-parent">
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/RecipeStepsEmptyScreen/index.tsx]: use styled components

File: app/cdap/components/RecipeSteps/RecipeStepsTableComponent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  TableRow,
+} from '@material-ui/core';
+import React from 'react';
+import { useStyles } from 'components/RecipeSteps/styles';
+import { IRecipeStepTebleProps } from 'components/RecipeSteps/RecipeStepsTableComponent/types';
+import { DeleteIcon } from 'components/RecipeSteps/iconStore';
+import { headerData } from 'components/RecipeSteps/RecipeStepsTableComponent/utils';
+
+export default function({ recipeSteps, handleDeleteRecipeSteps }) {
+  const classes = useStyles();
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/RecipeStepsTableComponent/index.tsx]: use Styled componets

File: app/cdap/components/RecipeSteps/RecipeStepsTableComponent/types.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export interface IRecipeStepTebleProps {
+  recipeSteps: unknown;
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/RecipeStepsTableComponent/types.ts]: remove unknown

File: app/cdap/components/RecipeSteps/iconStore.tsx [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import React from 'react';
+
+export const DownloadIcon = (
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/iconStore.tsx]: move it to IconStore

File: app/cdap/components/RecipeSteps/index.tsx [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+import DataPrepStore from 'components/DataPrep/store';
+import DrawerWidget from 'components/DrawerWidget';
+import React, { useEffect, useState } from 'react';
+import RecipeHeaderActionTemplate from './RecipeHeaderActionTemplate';
+import RecipeStepsEmptyScreen from './RecipeStepsEmptyScreen';
+import RecipeStepsTableComponent from './RecipeStepsTableComponent';
+import { useStyles } from './styles';
+import T from 'i18n-react';
+import { IRecipesSteps, IRecipeStepsProps } from './types';
+import { IRecipeStepTebleProps } from './RecipeStepsTableComponent/types';
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/index.tsx]: should not use relative paths

File: app/cdap/components/RecipeSteps/index.tsx [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    description: "'Body'",
+  },
+];
+
+const recipe_steps = ['uppercase: body1', 'titlecase: body2'];
+
+export default function({ setShowRecipePanel, showRecipePanel, deleteRecipes }) {
+  const [recipeSteps, setRecipeSteps] = useState(recipes);
+
+  const classes = useStyles();
```
Comment (dlakshmi123) [inline app/cdap/components/RecipeSteps/index.tsx]: use styled components

File: app/cdap/components/Snackbar/Components/Transition/index.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+export default function({
+  handleClose,
+  isSuccess,
+  messageToDisplay,
+  transitionAction,
+}: ITransitionProps) {
+  const handleUndoOperation = () => {
+    // TODO: this is the method used to undo the recent activity on transformations
+  };
+  const classes = useStyles();
```
Comment (dlakshmi123) [inline app/cdap/components/Snackbar/Components/Transition/index.tsx:36]: use styled componnets

File: app/cdap/components/Snackbar/index.tsx [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+import { ISnackbarProps } from 'components/Snackbar/types';
+import React, { useEffect, useState } from 'react';
+
+export default function({
+  handleCloseError,
+  description = '',
+  isSuccess,
+  snackbarAction,
+}: ISnackbarProps) {
+  const classes = useStyles();
```
Comment (dlakshmi123) [inline app/cdap/components/Snackbar/index.tsx:29]: use styled components

File: app/cdap/components/DrawerWidget/DrawerWidgetHeading/index.tsx (hunk 1 above)
Comment (dlakshmi123) [reply app/cdap/components/DrawerWidget/DrawerWidgetHeading/index.tsx]: done

File: app/cdap/components/DrawerWidget/iconStore.tsx (hunk 2 above)
Comment (dlakshmi123) [reply app/cdap/components/DrawerWidget/iconStore.tsx]: done

File: app/cdap/components/GridTable/index.tsx (hunk 5 above)
Comment (dlakshmi123) [reply app/cdap/components/GridTable/index.tsx]: done

File: app/cdap/components/GridTable/index.tsx (hunk 7 above)
Comment (dlakshmi123) [reply app/cdap/components/GridTable/index.tsx]: done

File: app/cdap/components/RecipeSteps/RecipeHeaderActionTemplate/index.tsx (hunk 8 above)
Comment (dlakshmi123) [reply app/cdap/components/RecipeSteps/RecipeHeaderActionTemplate/index.tsx]: done

File: app/cdap/components/RecipeSteps/RecipeStepsEmptyScreen/index.tsx (hunk 9 above)
Comment (dlakshmi123) [reply app/cdap/components/RecipeSteps/RecipeStepsEmptyScreen/index.tsx]: done

File: app/cdap/components/RecipeSteps/RecipeStepsTableComponent/index.tsx (hunk 10 above)
Comment (dlakshmi123) [reply app/cdap/components/RecipeSteps/RecipeStepsTableComponent/index.tsx]: done

File: app/cdap/components/RecipeSteps/RecipeStepsTableComponent/types.ts (hunk 11 above)
Comment (dlakshmi123) [reply app/cdap/components/RecipeSteps/RecipeStepsTableComponent/types.ts]: done

File: app/cdap/components/DrawerWidget/index.tsx (hunk 3 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/DrawerWidget/index.tsx]: Done

**Pull request closed without merging**